frontend/viewer/src/lib/DictionaryEntry.svelte

```diff
@@ -12,47 +12,44 @@
   ] as const;
 </script>
 <script lang="ts">
-  import { derived } from 'svelte/store';
-
   import type { IEntry } from '$lib/dotnet-types';
-  import { headword } from './utils';
-  import {useWritingSystems} from './writing-systems';
+  import {useWritingSystemService} from './writing-system-service';
   import { usePartsOfSpeech } from './parts-of-speech';
 
   export let entry: IEntry;
   export let lines: number = 0;
 
   $: lines = entry.senses.length > 1 ? entry.senses.length + 1 : 1;
 
-  const allWritingSystems = useWritingSystems();
-  const wsColor = derived(allWritingSystems, ({vernacular, analysis}) => {
+  const writingSystemService = useWritingSystemService();
+  const wsColor = (() => {
     const vernacularColor: Record<string, typeof vernacularColors[number]> = {};
     const analysisColor: Record<string, typeof analysisColors[number]> = {};
-    vernacular.forEach((ws, i) => {
+    writingSystemService.vernacular.forEach((ws, i) => {
       vernacularColor[ws.wsId] = vernacularColors[i % vernacularColors.length];
     });
-    analysis.forEach((ws, i) => {
+    writingSystemService.analysis.forEach((ws, i) => {
       analysisColor[ws.wsId] = analysisColors[i % analysisColors.length];
     });
     return (ws: string, type: 'vernacular' | 'analysis') => {
       const colors = type === 'vernacular' ? vernacularColor : analysisColor;
       return colors[ws];
     };
-  });
+  })();
```

Comment:
I don't really love this, though I see how you got here from what you had before. Could you just move this whole block inside the WS Service and just expose a `wsColor` function? maybe you compute the `*Color` fields on the service constructor?

Reply:
Sounds good 👍